### PR TITLE
[testing] Acceptance tests for rhoas_kafka

### PR DESCRIPTION
**Pull request by juandspy**
### Summary

Added some acceptance tests for `rhoas_kafka`:

- `TestAccRHOASKafka_Basic`: checks that we are able to spin up a Kafka cluster and delete it.
- `TestAccRHOASKafka_Update`: checks that we are able to spin up a Kafka cluster, change its name and delete it. 
    - This one is failing because updating the name makes the resource to be replaced.
    - This is because the update function is not yet implemented.
    - I left this one commented waiting for the update feature to come.
- `TestAccRHOASKafka_Error`: checks that an error is returned if the configuration is wrong. For example, if the cloud provider isn't supported.

### Testing

Run `make testacc`. You have to export your `OFFLINE_TOKEN` variable first.

### CODE REVIEW

File: rhoas/kafkas/resource_kafka.go [hunk 1]
```diff
@@ -139,7 +143,9 @@ func kafkaDelete(ctx context.Context, d *schema.ResourceData, m interface{}) dia
 
 	_, err = deleteStateConf.WaitForStateContext(ctx)
 	if err != nil {
-		return diag.FromErr(errors.Wrapf(err, "Error waiting for example instance (%s) to be deleted", d.Id()))
+		if !strings.Contains(err.Error(), "not found") {
+			return diag.FromErr(errors.Wrapf(err, "Error waiting for example instance (%s) to be deleted", d.Id()))
+		}
```
Comment (jackdelahunt) [inline rhoas/kafkas/resource_kafka.go:148]: Maybe I am missing something here but why do we not care about the error if it contains "not found"

File: rhoas/kafkas/resource_kafka.go [hunk 2]
```diff
@@ -122,6 +123,9 @@ func kafkaDelete(ctx context.Context, d *schema.ResourceData, m interface{}) dia
 		Refresh: func() (interface{}, string, error) {
 			data, resp, err1 := api.KafkaMgmt().GetKafkaById(ctx, d.Id()).Execute()
 			if err1 != nil {
+				if err1.Error() == "404 Not Found" {
+					return data, "404", nil
+				}
 				if apiErr := utils.GetAPIError(resp, err); apiErr != nil {
```
Comment (jackdelahunt) [inline rhoas/kafkas/resource_kafka.go]: Need to pass err1 here to GetAPIError not err

File: rhoas/kafkas/resource_kafka.go [hunk 3]
```diff
@@ -122,6 +123,9 @@ func kafkaDelete(ctx context.Context, d *schema.ResourceData, m interface{}) dia
 		Refresh: func() (interface{}, string, error) {
 			data, resp, err1 := api.KafkaMgmt().GetKafkaById(ctx, d.Id()).Execute()
 			if err1 != nil {
+				if err1.Error() == "404 Not Found" {
+					return data, "404", nil
+				}
```
Comment (jackdelahunt) [inline rhoas/kafkas/resource_kafka.go:128]: Why are we checking this just not sure? Also we will do more error handling in the future anyway so nothing needs to be perfect right now.

File: rhoas/kafkas/resource_kafka.go (hunk 1 above)
Comment (juandspy) [reply rhoas/kafkas/resource_kafka.go:148]: I found that sometimes the API returns an error "not found" when the resource has been deleted and the terraform provider fails. So I thought that a "not found" is OK if we are deleting the resource

File: rhoas/kafkas/resource_kafka.go (hunk 2 above)
Comment (juandspy) [reply rhoas/kafkas/resource_kafka.go]: true, updated

File: rhoas/kafkas/resource_kafka.go (hunk 3 above)
Comment (juandspy) [reply rhoas/kafkas/resource_kafka.go:128]: Similarly to [this comment](https://github.com/redhat-developer/terraform-provider-rhoas/pull/24#discussion_r982093684), I discovered that sometimes the API returns a "404 Not Found" and the delete function errors. I think that if we get a 404, there shouldn't be any error, as not finding the resource means it has been deleted.

File: rhoas/kafkas/resource_kafka.go (hunk 1 above)
Comment (jackdelahunt) [reply rhoas/kafkas/resource_kafka.go:148]: Okay this is kind of weird but maybe there is no way around it.

File: rhoas/kafkas/resource_kafka.go (hunk 3 above)
Comment (jackdelahunt) [reply rhoas/kafkas/resource_kafka.go:128]: Cool, bit weird but as I said not sure how to get around it right now.

**Review by jackdelahunt — APPROVED** [review]
👍

**Pull request merged**